### ISSUE TITLE
Fix demand forecasting bugs to run on customer data

### ISSUE THREAD

**Issue by benitoThree** [ThirdAILabs/Universe#351]
# Description
Currently, each experimental run introduces a new error. I want to get our model to a state where we no longer worry about its correctness and we can start focusing on feature engineering and experimenting quickly.

MLFlow page: http://deplo-mlflo-15qe25sw8psjr-1d20dd0c302edb1f.elb.us-east-1.amazonaws.com/#/experiments/8

Time estimate: 4 days.

# Success criteria
- [x] Runs without errors (no NaN, no floating point error, etc)
- [x] The model learns somewhat consistently; the error mostly decreases with more epochs of learning
- [x] Final error < 90%
- [x] Proper ML Flow logging
- [ ] Up to date with main branch

# Update
* Realized that I had a typo when specifying loss function and it defaults to categorical cross entropy loss, so it makes sense that the results were horrible at first.
* Switched to WMAPE since that's what CommerceIQ wants. No more NaN. Got floating point error but fixed it.
* Error does not consistently decrease after each epoch. Some epochs, the training error is around 30% - 40%, other epochs the training error is 400%. Test errors range from 0% to 3500%. 
* Experiment are currently run featurizers branch, which branched off from main a long time ago. I want to merge the changes so we can use the most up-to-date API.
* First run results on MLFlow: http://deplo-mlflo-15qe25sw8psjr-1d20dd0c302edb1f.elb.us-east-1.amazonaws.com/#/experiments/8/runs/ac903db59d5a43efbf45f97ad3a8d845
* Seems like it's learning. Between epochs 2-6, both train and test errors fall somewhat consistently to 22% and 11% respectively. The big issue is that test error jumps up to 800% and train error jumps to 150%, before they both fall to 0, which is impossible, so something must be wrong.
* Ran again, and it seems like there's no learning this time, graph [here](http://deplo-mlflo-15qe25sw8psjr-1d20dd0c302edb1f.elb.us-east-1.amazonaws.com/#/metric/test_wmape?runs=[%2224894bc7ee9e4d1e8721df1e537ba731%22]&experiment=8&plot_metric_keys=[%22test_wmape%22]&plot_layout={%22autosize%22:true,%22xaxis%22:{},%22yaxis%22:{}}&x_axis=relative&y_axis_scale=linear&line_smoothness=1&show_point=false&deselected_curves=[]&last_linear_y_axis_range=[])
* Fixed the metric calculation, ran again, and it's clearly not learning, in fact it gets worse by the epoch, graph [here](http://deplo-mlflo-15qe25sw8psjr-1d20dd0c302edb1f.elb.us-east-1.amazonaws.com/#/metric/test_wmape?runs=[%2281f2b8c417b848f98273c192e931832b%22]&experiment=8&plot_metric_keys=[%22test_wmape%22]&plot_layout={%22autosize%22:true,%22xaxis%22:{},%22yaxis%22:{}}&x_axis=relative&y_axis_scale=linear&line_smoothness=1&show_point=false&deselected_curves=[]&last_linear_y_axis_range=[])
* Fixed the loss function, ran again, and now it doesn't get worse over time but stays at the same absurdly high error (like 2,200% training error), eventually giving NaN, graph [here](http://deplo-mlflo-15qe25sw8psjr-1d20dd0c302edb1f.elb.us-east-1.amazonaws.com/#/metric/test_wmape?runs=[%227904f8f7521542cd948e4a8cdfa01bf9%22]&experiment=8&plot_metric_keys=[%22test_wmape%22]&plot_layout={%22autosize%22:true,%22xaxis%22:{},%22yaxis%22:{}}&x_axis=relative&y_axis_scale=linear&line_smoothness=1&show_point=false&deselected_curves=[]&last_linear_y_axis_range=[])
* Wrote tests for metric and loss function, and they seem to work fine; a trivial dataset gets optimized to close to 0% error, and even a randomly generated dataset (random examples, independently random labels) gets < 100% training wmape.
- [ ] Next step: reimplement wmape with changes from main
- [ ] Next step: reimplement featurizers with changes from main
- [ ] Next step: compare with the averaging model
